### ISSUE TITLE
Use of testing data in training phase

### ISSUE THREAD

**Issue by m-shayan73** [threaTrace-detector/threaTrace#18]
Hello, I have a concern about potential overlap between validation and testing sets.

In streamspot, Validation Set A has 20 random graphs from graph # 300-399 and Testing Set of attack graphs have 10 random graphs from graph # 300-399. Validation Set B has 10 random graphs from each set of 100 graphs (0-99,100,199,...) and Testing Set of benign graphs has the (remaining) 25 graphs from each set of 100 graphs. Here, the validation and testing set can include the same graphs.

Similarly in unicorn, Validate A has 10 attack graphs, all of which are in the test set and Validate B has 50 benign graphs, which could be in the test set.

In darpa, the test file is being accessed and any model that performs poorly is deleted with the condition
if (_tp/len(nodeA) > 0.8) and (_tp/(_tp+_fp+eps) > 0.7)

In my understanding, this will cause data leakage and prevent the system from generalising well. Would really appreciate any comment on this.

**Comment by SharedyX** [comment]
i have same concern. Do you solve this problem?
